### PR TITLE
CW + SDK v46 + ibcv3

### PR DESCRIPTION
**Pull request by faddat**
This draft PR is a part of the initiative described here:

https://twitter.com/gadikian/status/1496100856235950086?s=20&t=yIA9N83HtFoWVNCXKbagbw

and here:

https://github.com/cosmos/cosmos-sdk/issues/11096

Notes and questions:

* V46 refactors gov.  We are unsure if we should use new or legacy proposal types.  Currently we are using legacy.
* We're using the arm support branch of wasmvm, just to ease the development workflow.  This could be removed if need be. 
* ICA: I am not sure if there is integration work needed, where and how

### CODE REVIEW

**Comment by faddat** [comment]
@catShaark @nguyenvuong1122000 guys, I think that we'll want to use gov v1beta2 because otherwise we'll be using a bunch of deprecated code-- same thing goes for

File: x/wasm/keeper/test_common.go [hunk 1]
```diff
@@ -349,18 +363,17 @@ func createTestInput(
...
 	)
 
-	router := baseapp.NewRouter()
-	bh := bank.NewHandler(bankKeeper)
-	router.AddRoute(sdk.NewRoute(banktypes.RouterKey, bh))
-	sh := staking.NewHandler(stakingKeeper)
-	router.AddRoute(sdk.NewRoute(stakingtypes.RouterKey, sh))
-	dh := distribution.NewHandler(distKeeper)
-	router.AddRoute(sdk.NewRoute(distributiontypes.RouterKey, dh))
+	router := authmiddleware.NewLegacyRouter()
```
Comment (faddat) [inline x/wasm/keeper/test_common.go:366]: this stuff here.

File: x/wasm/keeper/test_common.go [hunk 2]
```diff
@@ -349,18 +363,17 @@ func createTestInput(
...
+	br := bank.AppModule.Route(bank.AppModule{})
+	router.AddRoute(br)
+	sr := staking.AppModule.Route(staking.AppModule{})
+	router.AddRoute(sr)
+	dr := distribution.AppModule.Route(distribution.AppModule{})
+	router.AddRoute(dr)
```
Comment (faddat) [inline x/wasm/keeper/test_common.go:372]: sorry, I meant all of this.  It is throwing deprecated errors from golangci-lint

**Comment by faddat** [comment]
@alpe no problem!  It will be a fairly beefy one and it'll come with questions for sure.

Basically, we ended up needing this and figured it made sense to contribute :)

**Comment by faddat** [comment]
To anyone watching this-- @catShaark  and I just met and we're going to do gov in phases:

1) v1beta1
2) v1beta2

**Comment by faddat** [comment]
**Status Update**

In order to make this work properly, Khanh has made a PR to ibc-go which will end up in the ibc spec:

https://github.com/cosmos/ibc-go/pull/1076

https://github.com/cosmos/ibc-go/issues/1075

https://github.com/cosmos/ibc/issues/674

From here, we are going to begin actively testing the work found here in [craft economy](https://github.com/notional-labs/craft)

The goal is basically the same as always-- to provide a featureful platform for blockchain development that tracks:

* cosmos-sdk
* wasmd
* ibc-go

And to create a templating tool that lives in the SDK, which ships with all features turned on so that it will be useful in a CI system, which will also land in the SDK repo.

**Comment by faddat** [comment]
Only problem remaining is the benchmarks and I suspect that's trivial.

**Comment by ethanfrey** [comment]
Hi @faddat I am just back from vacation and seeing this.

We do not plan to add v0.46 support to wasmd:main anytime soon, but do plan to add ibcv3 support soon.
If you want to pull out just the ibcv3 part of this into another PR, I can review it later this week. It would keep the v0.46 fork closer to the main branch.

**Comment by ethanfrey** [comment]
We won't work on v0.46 for some time. Not looking into this until then.

But I will review #793 shortly, which is compatible with main

**Pull request closed without merging**